client/hostinfo_unix.cpp

```diff
@@ -2003,6 +2010,75 @@ const vector<string> X_display_values_initialize() {
     return display_values;
 }
 
+void getIdleTime() {
+    long idleTime = 0;
+    // Use glob to enumerate input devices in /dev/input/
+    glob_t globbuf;
+    if (glob("/dev/input/event*", GLOB_NOSORT, nullptr, &globbuf) != 0) {
+        std::cerr << "Failed to enumerate input devices. " << std::endl;
+        return;
+    }
+
+    // Create libevdev structures for each device
+    libevdev* devices[globbuf.gl_pathc];
+ 
+    // Open and initialize each device
+    for (size_t i = 0; i < globbuf.gl_pathc; ++i) {
+        const char* devicePath = globbuf.gl_pathv[i];
+        int fd = open(devicePath, O_RDONLY | O_NONBLOCK);
+        if (fd < 0) {
+            std::cerr << "Failed to open device (Permission denied?): " << devicePath << std::endl;
+            return;
+        }
+
+        if (libevdev_new_from_fd(fd, &devices[i]) < 0) {
+            std::cerr << "Failed to initialize libevdev for device: " << devicePath << std::endl;
+            return;
+        }
+    }
+
+    // Main loop to monitor input events
+    while (true) {
```

Comment:
This will lead to the dead lock of the BOINC Client. Instead, this functions should be called outside and return result as fast as possible.
Please be sure to test your changes before making a PR.

Reply:
I want to have an opinion before making any further changes, This code could be added separately, runs in the background while the client is running, and writes the time of being idle to a tmp file, which can be read inside the client at any time.
But I am not sure if you agree with this idea to be added to the code, 

Also, as I mentioned it will need permissions to read and execute /dev/input which regular users don't have by default.
So separating it could be a better approach to give the file its own permissions because having such permissions to a whole user is a security risk.

Reply:
Don't get me wrong, but this is a not a good approach: you did some implementation, but you haven't tested it. What is more important - we can't test it as well, because we can't just use your code and run any test, because obviously everything will just hang in an infinite loop.
If you have an idea but don't know how to implement it properly - it's better to describe your idea in detail and ask for advice.

Reply:
You are right.
I've removed the infinite loop and fixed the stupid logic.
But I also haven't removed any existing code.

I'll push the updates to the pull (when I figure out how).

Reply:
While on this branch:
git add . && git commit && git push